### ISSUE TITLE
3-overlapErrorAdjustment segfaulting V1.9

### ISSUE THREAD

**Issue by pollend** [marbl/canu#1504]
so 3-overlapErrorAdjustment segfaulted. I'm using the V1.9 branch any clue how to fix this. I'm trying to rerun the step and see if it passes then rerun the whole thing.

this is using commit: 5476d4e5065a9038a0c21baddaa29286806cdc53

```
Running job 1 based on SLURM_ARRAY_TASK_ID=1 and offset=0.
Initializing.
Opening seqStore '../../<omitted>.seqStore'.
Correcting reads 1 to 13168.
Reading 137758 corrections from './red.red'.
Correcting 175758054 bases with 97923 indel adjustments.
--Allocate 167 + 0 + 0 MB for bases, adjusts and reads.
Corrected 175761015 bases with 26667 substitutions, 47481 deletions and 50442 insertions.
Loading overlaps.
Read_Olaps()--  Loading 32358526 overlaps from '../<omitted>.ovlStore' for reads 1 to 13168
--Allocate 987 MB for overlaps.
Read_Olaps()--  Loaded 32358526 overlaps -- 16204404 normal and 16154122 innie.
Sorting overlaps.
Recomputing overlaps.
--Allocate 8 MB for fseq and rseq.
--Allocate 32 MB for fadj and radj.
--Allocate 16 MB for pedWorkArea_t.
-- Allocate 0 MB for Edit_Array pointers.
--Allocate 16 MB for edit array work space 0 (positions 0-4092)
Recomputing overlaps -         1 -     12289 -     13167
--  Release bases, adjusts and reads.
*** Error in `/home/<omitted>/Linux-amd64/bin/correctOverlaps': munmap_chunk(): invalid pointer: 0x00002aaab8723010 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x7f5e4)[0x2aaaaba325e4]
/home/<omitted>/canu/Linux-amd64/bin/correctOverlaps[0x408084]
/home/<omitted>/canu/Linux-amd64/bin/correctOverlaps[0x403009]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x2aaaab9d5445]
/home/<omitted>/canu/Linux-amd64/bin/correctOverlaps[0x4036b7]
======= Memory map: ========
00400000-0043a000 r-xp 00000000 00:26 107385570040                       /home/<omitted>/Linux-amd64/bin/correctOverlaps
0063a000-0063b000 r--p 0003a000 00:26 107385570040                       /home/<omitted>/Linux-amd64/bin/correctOverlaps
0063b000-0063c000 rw-p 0003b000 00:26 107385570040                       /home/<omitted>/canu/Linux-amd64/bin/correctOverlaps
0063c000-007aa000 rw-p 00000000 00:00 0                                  [heap]
2aaaaaaab000-2aaaaaacd000 r-xp 00000000 09:01 53184736                   /usr/lib64/ld-2.17.so
2aaaaaacd000-2aaaaaacf000 r-xp 00000000 00:00 0                          [vdso]
2aaaaaacf000-2aaaaaad6000 rw-p 00000000 00:00 0 
2aaaaaae7000-2aaaaab6f000 rw-p 00000000 00:00 0 
2aaaaaccc000-2aaaaaccd000 r--p 00021000 09:01 53184736                   /usr/lib64/ld-2.17.so
2aaaaaccd000-2aaaaacce000 rw-p 00022000 09:01 53184736                   /usr/lib64/ld-2.17.so
2aaaaacce000-2aaaaaccf000 rw-p 00000000 00:00 0 
2aaaaaccf000-2aaaaae41000 r-xp 00000000 09:01 2149546021                 /cm/local/apps/gcc/7.2.0/lib64/libstdc++.so.6.0.24
2aaaaae41000-2aaaab040000 ---p 00172000 09:01 2149546021                 /cm/local/apps/gcc/7.2.0/lib64/libstdc++.so.6.0.24
2aaaab040000-2aaaab04a000 r--p 00171000 09:01 2149546021                 /cm/local/apps/gcc/7.2.0/lib64/libstdc++.so.6.0.24
2aaaab04a000-2aaaab04c000 rw-p 0017b000 09:01 2149546021                 /cm/local/apps/gcc/7.2.0/lib64/libstdc++.so.6.0.24
2aaaab04c000-2aaaab050000 rw-p 00000000 00:00 0 
2aaaab050000-2aaaab151000 r-xp 00000000 09:01 53258034                   /usr/lib64/libm-2.17.so
2aaaab151000-2aaaab350000 ---p 00101000 09:01 53258034                   /usr/lib64/libm-2.17.so
2aaaab350000-2aaaab351000 r--p 00100000 09:01 53258034                   /usr/lib64/libm-2.17.so
2aaaab351000-2aaaab352000 rw-p 00101000 09:01 53258034                   /usr/lib64/libm-2.17.so
2aaaab352000-2aaaab37f000 r-xp 00000000 09:01 2149545386                 /cm/local/apps/gcc/7.2.0/lib64/libgomp.so.1.0.0
2aaaab37f000-2aaaab57e000 ---p 0002d000 09:01 2149545386                 /cm/local/apps/gcc/7.2.0/lib64/libgomp.so.1.0.0
2aaaab57e000-2aaaab57f000 r--p 0002c000 09:01 2149545386                 /cm/local/apps/gcc/7.2.0/lib64/libgomp.so.1.0.0
2aaaab57f000-2aaaab580000 rw-p 0002d000 09:01 2149545386                 /cm/local/apps/gcc/7.2.0/lib64/libgomp.so.1.0.0
2aaaab580000-2aaaab596000 r-xp 00000000 09:01 2149545379                 /cm/local/apps/gcc/7.2.0/lib64/libgcc_s.so.1
2aaaab596000-2aaaab795000 ---p 00016000 09:01 2149545379                 /cm/local/apps/gcc/7.2.0/lib64/libgcc_s.so.1
2aaaab795000-2aaaab796000 r--p 00015000 09:01 2149545379                 /cm/local/apps/gcc/7.2.0/lib64/libgcc_s.so.1
2aaaab796000-2aaaab797000 rw-p 00016000 09:01 2149545379                 /cm/local/apps/gcc/7.2.0/lib64/libgcc_s.so.1
2aaaab797000-2aaaab7ae000 r-xp 00000000 09:01 53305520                   /usr/lib64/libpthread-2.17.so
2aaaab7ae000-2aaaab9ad000 ---p 00017000 09:01 53305520                   /usr/lib64/libpthread-2.17.so
2aaaab9ad000-2aaaab9ae000 r--p 00016000 09:01 53305520                   /usr/lib64/libpthread-2.17.so
2aaaab9ae000-2aaaab9af000 rw-p 00017000 09:01 53305520                   /usr/lib64/libpthread-2.17.so
2aaaab9af000-2aaaab9b3000 rw-p 00000000 00:00 0 
2aaaab9b3000-2aaaabb76000 r-xp 00000000 09:01 53193677                   /usr/lib64/libc-2.17.so
2aaaabb76000-2aaaabd75000 ---p 001c3000 09:01 53193677                   /usr/lib64/libc-2.17.so
2aaaabd75000-2aaaabd79000 r--p 001c2000 09:01 53193677                   /usr/lib64/libc-2.17.so
2aaaabd79000-2aaaabd7b000 rw-p 001c6000 09:01 53193677                   /usr/lib64/libc-2.17.so
2aaaabd7b000-2aaaabd80000 rw-p 00000000 00:00 0 
2aaaabd80000-2aaaabd82000 r-xp 00000000 09:01 53228960                   /usr/lib64/libdl-2.17.so
2aaaabd82000-2aaaabf82000 ---p 00002000 09:01 53228960                   /usr/lib64/libdl-2.17.so
2aaaabf82000-2aaaabf83000 r--p 00002000 09:01 53228960                   /usr/lib64/libdl-2.17.so
2aaaabf83000-2aaaabf84000 rw-p 00003000 09:01 53228960                   /usr/lib64/libdl-2.17.so
2aaaabf84000-2aaaadf85000 rw-p 00000000 00:00 0 
2aaab8723000-2aaaf63cb000 rw-p 00000000 00:00 0 
7ffffffdc000-7ffffffff000 rw-p 00000000 00:00 0                          [stack]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]

Failed with 'Aborted'; backtrace (libbacktrace):
utility/system-stackTrace.C::89 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
overlapErrorAdjustment/correctOverlaps-Redo_Olaps.C::554 in _Z10Redo_OlapsP12coParametersP7sqStore()
overlapErrorAdjustment/correctOverlaps.C::171 in main()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
```

**Comment by skoren** [comment]
Are you sure you're on the 1.9 branch, what does canu -version report? It should say 1.9 in the version string. The commit you report is actually on the master not 1.9. 

Try switching to the 1.9 branch and re-building (which would require re-starting the assembly) or you could try updating to latest master and re-run this failing job.

**Comment by pollend** [comment]
```
[polli104@chapman.edu@bright81 canu]$ git branch 
  master
* v1.9
[polli104@chapman.edu@bright81 canu]$ git log
commit 5476d4e5065a9038a0c21baddaa29286806cdc53 (HEAD -> v1.9, origin/v1.9)
Author: Brian P. Walenz <thebri@gmail.com>
Date:   Thu Aug 15 09:31:15 2019 -0400

    Fix embarassing bug filtering out most if not all insertion corrections.  From 7c4cade2235c3d7d41806ea888efaf00d4287efb.

commit 503ea967e1acc9977794c27c82d8821658a1eb09
Author: Brian P. Walenz <thebri@gmail.com>
Date:   Wed Aug 14 16:09:20 2019 -0400

    Add check for array size on LSF clusters

commit 348be69d2ba8c7ec838b230ae26a58800097bf0d
Author: Brian P. Walenz <thebri@gmail.com>
Date:   Wed Aug 14 16:08:10 2019 -0400

    Skip calling lsadmin if gridEngineMemoryUnits is set.  Issue #1443.

commit c1ccd1ebe6b78f5181c3dc56b29281595a48f324
Author: Brian P. Walenz <thebri@gmail.com>
Date:   Wed Aug 14 15:24:18 2019 -0400

    Be consistent about case of 'SLURM'.

```

**Comment by skoren** [comment]
OK, thanks, I got confused because the same hash exists in multiple branches. There were some large changes to the OEA code since your version. Do an update, recompile, and see if the job runs. If it still does not work, would you be able to share the data with us to debug locally?

**Comment by pollend** [comment]
Master seems to have a lot problem so I was recommended V1.9. Is there specific branch or commit I should use instead. I would need to talk with advisor, but I don't think he will let me share the data.

**Comment by skoren** [comment]
I'm not suggesting using master, it's not guaranteed to be stable and shouldn't be used for production, but your 1.9 branch is also out of date and has had OEA code changes so if you pull on that branch and recompile it will give a better idea if the error is still present. 

Since you're not able to share the data, if it still crashes then your best option will be to go to the 1.8 release. You'll have to re-run the unitigging step of the assembly using ```canu -assemble -pacbio-corrected``` or ```-nanopore-corrected``` as appropriate for your data.

**Comment by skoren** [comment]
Idle, same error as #1509 so will be tracked there. Can't get data to reproduce and I expect it is already fixed. Best fix is to use latest 1.9 code or 1.8 release.

**Comment by pollend** [comment]
looks like it's working now with the newest 1.9 version of that branch.
